### PR TITLE
DOC: recommend polynomial.Polynomial over np.polyfit

### LINKED ISSUE #11732: numpy.polynomial.polynomial.polyfit and numpy.polyfit return coefficients in different order
### Summary

This may be a 'historical reasons' issue, but it looks like ``numpy.polynomial.polynomial.polyfit`` and ``numpy.polyfit`` return coefficients in reverse order from each other. I wonder if one of the functions should be deprecated from the public API in future, as having two functions with the same name in the same package that operate differently can lead to confusion?

### Reproducing code example:

```python
import numpy as np
from numpy.polynomial.polynomial import polyfit

x = np.linspace(0, 1, 100)
y = x ** 3

p1 = np.polyfit(x, y, 3)
p2 = polyfit(x, y, 3)

print('p1', p1)
print('p2', p2)
```

### Result

```
p1 [ 1.00000000e+00  4.64782291e-16 -7.05817953e-16 -1.47054969e-17]
p2 [-1.93895090e-16  8.46391197e-16 -7.28660556e-16  1.00000000e+00]
```

### Numpy/Python version information:

Python 3.6, Numpy 1.15.0, installed with conda.

### PR DESCRIPTION
**Pull request by mattip**
Closes #11732 as per the comment in the issue.

### CODE REVIEW

File: numpy/lib/polynomial.py [hunk 1]
```diff
@@ -398,6 +398,11 @@ def polyfit(x, y, deg, rcond=None, full=False, w=None, cov=False):
     to points `(x, y)`. Returns a vector of coefficients `p` that minimises
     the squared error.
 
+    Please use `numpy.polynomial.Polynomial.fit` for new code. This remains
```
Comment (mhvk) [inline numpy/lib/polynomial.py]: Should we refer to `np.polynomial.polynomial.polyfit` instead? Or is the class method better?

Comment (mhvk) [reply numpy/lib/polynomial.py]: Never mind, I now saw @charris's comment that in fact the class method is recommended.

Comment (charris) [reply numpy/lib/polynomial.py]: The class method is numerically better, but to get the conventional coefficients one needs to do `p.convert()` on the resulting `p`.

Comment (mattip) [reply numpy/lib/polynomial.py]: Could you suggest an alternative wording?

Comment (charris) [reply numpy/lib/polynomial.py]: Maybe,
"The `numpy.polynomial.Polynomial.fit` class method is recommended for new code as it is more stable numerically. See the documentation for the method for more information."

We might want to add a mention to Polynomial documentation about how to retrieve the conventional coefficients in the `Returns` section. The documentation is located in `numpy/polynomial/_polybase.py`.

**Comment by mattip** [comment]
Added a comment to the `return` section of `Polynomial.fit`. Examples are lacking in that class.

File: numpy/polynomial/_polybase.py [hunk 2]
```diff
@@ -860,7 +860,8 @@ class domain in NumPy 1.4 and ``None`` in later versions.
         -------
         new_series : series
             A series that represents the least squares fit to the data and
-            has the domain specified in the call.
+            has the domain specified in the call. ``newseries.coef`` will
```
Comment (charris) [inline numpy/polynomial/_polybase.py]: Needs to run the convert method to see the coefficients in standard form. So maybe
```
has the domain and window specified in the call. If the coefficients for the unscaled and unshifted
basis polynomials are of interest, do ``new_series.convert().coef`` .
```

Comment (mattip) [reply numpy/polynomial/_polybase.py]: adopted

File: numpy/polynomial/_polybase.py [hunk 3]
```diff
@@ -860,7 +860,10 @@ class domain in NumPy 1.4 and ``None`` in later versions.
         -------
         new_series : series
             A series that represents the least squares fit to the data and
-            has the domain specified in the call.
+            has the domain and window specified in the call. If the
+            coefficients for the unscaled and unshifted basis polynomials are
+            of interest, do `new_series.convert().coef
+            <numpy.polynomial.polynomial.Polynomial.convert>`
```
Comment (charris) [inline numpy/polynomial/_polybase.py]: I wonder if it is worth making links in the documentation? Just seems to make it harder to read in the console. So maybe just ``` ``new_series.convert().coef`` ```

Comment (charris) [reply numpy/polynomial/_polybase.py]: Also note that this is a base class and is used for six different polynomial classes, so the documentation needs to be generic.

**Comment by charris** [comment]
Thanks Matti.

File: numpy/lib/polynomial.py [hunk 4]
```diff
@@ -396,7 +396,12 @@ def polyfit(x, y, deg, rcond=None, full=False, w=None, cov=False):
 
     Fit a polynomial ``p(x) = p[0] * x**deg + ... + p[deg]`` of degree `deg`
     to points `(x, y)`. Returns a vector of coefficients `p` that minimises
-    the squared error.
+    the squared error in the order `deg`, `deg-1`, ... `0`.
+
+    The `Polynomial.fit <numpy.polynomial.polynomial.Polynomial.fit>` class
+    method is recommended for new code as it is more stable numerically. See
+    the documentation for the method for more information, or the convenience
+    function `polynomial.polyfit <numpy.polynomial.polynomial.polyfit>`.
```
[the remark this answers is not in the document]
Comment (charris) [reply numpy/lib/polynomial.py]: Oops, no, it isn't. That last clause came out of the blue and needs to be removed, as the implementation function `polyfit` doesn't have documentation useful for the class function.

**Pull request merged**